src/sync.rs

```diff
@@ -31,10 +34,12 @@ use std::cell::UnsafeCell;
 ///     NUMBERS.get(py).borrow_mut().push(42);
 /// });
 /// ```
+#[cfg(not(Py_GIL_DISABLED))]
```

Comment:
I think we'll need a `4504.changed.md` newsfragment to document the unavailability of this API on the freethreaded build.